src/utils.cpp

```diff
@@ -177,20 +177,40 @@ std::vector<std::string> utils::tokenize_spaced(const std::string& str, std::str
 	std::string::size_type pos = str.find_first_of(delimiters, last_pos);
 
 	if (last_pos != 0) {
-		tokens.push_back(std::string(" "));
+		tokens.push_back(str.substr(0, last_pos));
 	}
 
 	while (std::string::npos != pos || std::string::npos != last_pos) {
 		tokens.push_back(str.substr(last_pos, pos - last_pos));
 		last_pos = str.find_first_not_of(delimiters, pos);
 		if (last_pos > pos)
-			tokens.push_back(std::string(" "));
+			tokens.push_back(str.substr(pos, last_pos - pos));
 		pos = str.find_first_of(delimiters, last_pos);
 	}
 
 	return tokens;
 }
 
+std::string utils::consolidate_whitespace(const std::string& str, std::string whitespace) {
+	std::string result;
+	std::string::size_type last_pos = str.find_first_not_of(whitespace, 0);
+	std::string::size_type pos = str.find_first_of(whitespace, last_pos);
+
+	if(last_pos != 0){
```

Comment:
Don't forget the space after the keyword! And a space before the curly bracket, too.